frontend/pulse3d/src/components/layouts/ControlPanel.js

```diff
@@ -118,82 +103,148 @@ const adminButtons = [
   },
   { label: "Users Info", disabled: false, page: "/users-info", options: [] },
 ];
+const modalObjs = {
+  jobsReached: {
+    header: "Warning!",
+    messages: [
+      "All usage limits have been reached for this customer account.",
+      "Users will not be able to upload new recording files or perform re-analysis on existing files.",
+    ],
+  },
+  uploadsReached: {
+    header: "Warning!",
+    messages: [
+      "The upload limit has been reached for this customer account.",
+      "Users will only be allowed to perform re-analysis on existing files.",
+    ],
+  },
+};
 
 export default function ControlPanel() {
   const router = useRouter();
   const [selected, setSelected] = useState("Home");
   const [expanded, setExpanded] = useState(null);
-  const { accountType } = useContext(AuthContext);
+  const { accountType, usageQuota } = useContext(AuthContext);
+  const [modalState, setModalState] = useState(false);
+  const [modalLabels, setModalLabels] = useState({ header: "", messages: [] });
+
+  const userButtons = [
+    { label: "Home", disabled: false, page: "/uploads", options: [] },
+    {
+      label: "Run Analysis",
+      disabled: usageQuota && usageQuota.jobs_reached, // disabled completely if jobs quota has been reached
+      page: "/upload-form",
+      options:
+        usageQuota && usageQuota.uploads_reached // prevent new analyses if uploads quota reached
+          ? ["Re-analyze Existing Upload"]
+          : ["Analyze New Files", "Re-analyze Existing Upload"],
+    },
+    {
+      label: "Account Settings",
+      disabled: true,
+      page: "/account",
+      options: [],
+    },
+  ];
+
   const buttons = accountType === "admin" ? adminButtons : userButtons;
 
   useEffect(() => {
+    // this checks if a page changes without button is clicked from a forced redirection
     const currentPage = buttons.filter(({ page }) => page === router.pathname)[0];
 
     if (currentPage) {
       const { label, options } = currentPage;
       if (label !== selected) setSelected(label);
       if (options.length > 0) setExpanded(label);
+      else if (options.length === 0) setExpanded(null);
```

Comment:
This could just be `else setExpanded(null);` unless `options.length` can be < 0 somehow